includes/class-amp-reader-themes.php

```
@@ -0,0 +1,546 @@
+		 *         @type string         $description     A description of the theme.
+		 *         @type string|boolean $requires        Minimum version of WordPress required by the theme. False if all versions are supported.
+		 *         @type string|boolean $requires_php    Minimum version of PHP required by the theme. False if all versions are supported.
+		 *         @type string         $download_link   A link to the theme's zip file. If empty, the plugin will attempt to download the theme from wordpress.org.
+		 *     }
+		 * ]
+		 */
+		$themes = apply_filters( 'amp_reader_themes', $themes );
+
+		foreach ( $themes as &$theme ) {
+			$theme['availability'] = $this->get_theme_availability( $theme );
+		}
+
+		$this->themes = $themes;
+
+		return $this->themes;
+	}
+
+	/**
+	 * Gets a reader theme by slug.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $slug Theme slug.
+	 * @return array Theme data.
+	 */
+	public function get_reader_theme( $slug ) {
+		return current(
+			array_filter(
+				$this->get_themes(),
+				static function( $theme ) use ( $slug ) {
+					return $theme['slug'] === $slug;
+				}
+			)
+		);
+	}
+
+	/**
+	 * Retrieves theme data.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param boolean $from_api Whether to return theme data from the wordpress.org API. Default false.
+	 * @return array Theme ecosystem posts copied the amp-wp.org website.
+	 */
+	public function get_default_supported_reader_themes( $from_api = false ) {
+		// Note: This can be used to refresh the hardcoded raw theme data.
+		if ( $from_api ) {
+			if ( ! function_exists( 'themes_api' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/theme.php';
+			}
+
+			$response = themes_api(
+				'query_themes',
+				[
+					'author'   => 'wordpressdotorg',
+					'per_page' => 24, // There are only 12 as of 05/2020.
+				]
+			);
+
+			if ( ! $response || is_wp_error( $response ) ) {
+				return $response;
```

The method claims that it returns an array, but here we return `false` or `WP_Error` instead.
Either the typehint and the consuming code need to be adapted to deal with these types as well, or we should just silently fail and return an empty array here.